app/NoteAndAccidentalInput/page.tsx

```diff
@@ -1,162 +1,97 @@
 "use client";
-import React, { useEffect, useRef, useState } from "react";
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useReducer,
+  useRef,
+  useState,
+} from "react";
+import VexFlow from "vexflow";
 import BlueButton from "../components/BlueButton";
 import CheckIfNoteFound from "../components/CheckIfNoteFound";
 import CheckNumBeatsInMeasure from "../components/CheckNumBeatsInMeasure";
-import KaseyBlankStaves from "../components/KaseyBlankStaves";
-import { addAccidentalToNote } from "../lib/addAccidentalToNote";
+import {
+  clearAllMeasures,
+  modifyStaveNotesButtonGroup,
+} from "../lib/buttonsAndButtonGroups";
+import { staveData } from "../lib/data/stavesData";
 import { findBarIndex } from "../lib/findBar";
 import generateYMinAndYMaxForAllNotes from "../lib/generateYMinAndMaxForAllNotes";
-import { indexOfNoteToModify } from "../lib/indexOfNoteToModify";
+import getUserClickInfo from "../lib/getUserClickInfo";
+import { initializeRenderer } from "../lib/initializeRenderer";
+import { handleNoteInteraction } from "../lib/handleNoteInteraction";
 import { notesArray } from "../lib/noteArray";
-
+import { noteInteractionReducer } from "../lib/reducers";
+import { setupRendererAndDrawNotes } from "../lib/setupRendererAndDrawNotes";
 import {
-  NoteStringYMinAndYMaxUserClickY,
-  StateType,
-  StaveNoteAbsoluteXCoordUserClickY,
-  StaveNoteType,
+  NoteStringData,
+  StaveNoteData,
   StaveType,
 } from "../lib/typesAndInterfaces";
+import { noteInteractionInitialState } from "../lib/initialStates";
+const { Renderer } = VexFlow.Flow;
+import { INITIAL_STAVES } from "../lib/data/stavesData";
```

Comment:
Small thing: You can do a shift-option-o to quickly set the proper order of imports. You can do this for each file if you haven't already.